File: Projet_LO52/app/src/main/java/com/dayetfracso/codep25/dao/TeamDao.java
package com.dayetfracso.codep25.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Transaction;


import com.dayetfracso.codep25.entity.Runner;
import com.dayetfracso.codep25.entity.Team;
import com.dayetfracso.codep25.entity.TeamWithRunners;

import java.util.List;

@Dao
public interface TeamDao {
    @Insert
    long insertTeam(Team team);

    @Transaction
    @Query("SELECT * FROM teams")
    public List<Team> getAllTeams();

    @Transaction
    @Query("DELETE FROM teams")
    public void deleteAllTeams();

    @Transaction
    @Query("SELECT * FROM teams")
    public List<TeamWithRunners> getTeamsWithRunners();

    @Transaction
    @Query("SELECT * FROM teams where teamId = :teamId")
    public Team getTeam(long teamId);

    @Transaction
    @Query("SELECT * FROM teams where teamId = :teamId")
    public TeamWithRunners getTeamWithRunners(long teamId);
}

File: Projet/app/app/src/main/java/com/codep25/codep25/model/storage/dao/TeamDao.kt
package com.codep25.codep25.model.storage.dao

import androidx.room.*
import com.codep25.codep25.model.entity.Team
import com.codep25.codep25.model.entity.TeamWithParticipants

@Dao
interface TeamDao {

    @Query("SELECT * FROM teams")
    @Transaction
    fun getTeams(): List<TeamWithParticipants>

    @Query("SELECT * FROM teams WHERE id_team == :id")
    fun getTeamById(id: Long): TeamWithParticipants

    @Insert
    fun insert(team: Team): Long

    @Insert
    fun insertAll(vararg team: Team)

    @Update
    fun update(team: Team)

    @Delete
    fun delete(team: Team)

    @Query("DELETE FROM teams")
    fun clearTeams()
}